docs/authentication.md

```diff
@@ -0,0 +1,130 @@
+# A Guide on ScalarDL Authentication
+
+This document explains the ScalarDL authentication mechanism and how to use it properly. 
+
+## Authentication in ScalarDL
+
+Authentication is one of the key roles in ScalarDL to make the protocol work as expected.
+ScalarDL uses authentication for the following three places:
+* Client authentication (for Ledger and Auditor)
+    * Ledger and Auditor authenticates clients using client-generated signatures attached to requests from the clients.
```

Comment:
For the `by`, I noticed you fixed in other docs, but is it grammatically incorrect? I think `by` is kind of omitted here.
Or, you suggest this to make it more explicit? I'm not arguing and just would like to learn English more. 😄 

I found a similar discussion here.
https://forum.wordreference.com/threads/omit-the-preposition-by.3055198/

Reply:
Good question! "Using" is a tricky word because it can cause ambiguity, especially in technical content.

For example, the original sentence that uses `using` could mean either of the following:

* **Meaning 1:** Ledger and Auditor authenticate clients _that use_ client-generated signatures attached to requests from the clients.
  * **Ambiguous point:** How are those clients authenticated?
* **Meaning 2:** Ledger and Auditor authenticate clients _by using_ client-generated signatures attached to requests from the clients.
  * **Alternative reading:** By using client-generated signatures attached to requests from the clients, Ledger and Auditor authenticate those clients.

**Reference:** [using, by using, with - Microsoft Style Guide](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/u/using-by-using-with)